packages/rum/src/performanceCollection.ts

```diff
@@ -99,49 +101,58 @@ export function startPerformanceCollection(lifeCycle: LifeCycle) {
   }
 }
 
-export function retrieveInitialDocumentResourceTiming() {
-  let timing: RumPerformanceResourceTiming
+export function retrieveInitialDocumentResourceTiming(callback: (timing: RumPerformanceResourceTiming) => void) {
+  runOnReadyState('interactive', () => {
```

Comment:
Other function name is `retrieveNavigationTimingWhenLoaded`, so I think it could be nice to also reflect in the function name that it is waiting for a specific page state.
However, it could be even clearer to move the `runOnReadyState` outside of these functions:

```
runOnReadyState('interactive', () => {
  retrieveInitialDocumentResourceTiming((timing) => {
    handleRumPerformanceEntry(lifeCycle, timing)
  })
}}

runOnReadyState('complete', () => {
  retrieveNavigationTiming((timing) => {
    handleRumPerformanceEntry(lifeCycle, timing)
  })
}}
```

Reply:
Waiting for the DOM to be ready or document to be loaded is an implementation detail and I think it's better to keep it in the function. In particular, `retrieveNavigationTiming` is waiting for the document load + an bit of time for the `loadEventEnd` to be defined: it would be hard to understand what's going on if those tasks were in two different places.

I'm fine with renaming the function to `retrieveInitialDocumentResourceTiming` to `retrieveInitialDocumentResourceTimingWhenDomReady` even if I don't see much value in it.

Reply:
ok, so maybe rename `retrieveNavigationTimingWhenLoaded` to `retrieveNavigationTiming` to be consistent in considering it as an implementation detail